GVFS/GVFS.Common/NamedPipes/QueuedPipeWriter.cs

```diff
@@ -0,0 +1,209 @@
+using System;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.IO.Pipes;
+using System.Threading;
+
+namespace GVFS.Common.NamedPipes
+{
+    public class QueuedPipeWriter : IDisposable
+    {
+        private static readonly TimeSpan PollingInterval = TimeSpan.FromSeconds(30);
```

Comment:
When the queue goes to sleep, there are two ways it gets woken up again:

1. a new message is posted,
2. 30 seconds pass.